src/ebay-video/video.tsx

```diff
@@ -0,0 +1,270 @@
+    onLoadError = () => {
+    },
+    onPlay = () => {
+    },
+    onReport = () => {
+    },
+    children,
+    ...rest
+}) => {
+    const [loaded, setLoaded] = useState<boolean>()
+    const [buffering, setBuffering] = useState<boolean>()
+    const [playing, setPlaying] = useState<boolean>()
+    const [failed, setFailed] = useState<boolean>()
+
+    const containerRef = useRef<HTMLDivElement>(null)
+    const videoRef = useRef<HTMLVideoElement>(null)
+    const playerRef = useRef<any>(null)
+    const uiRef = useRef(null)
+
+    const sources = filterByType(children, EbayVideoSource).map(({ props }) => props)
+
+    const handleError = (err: any) => {
+        setLoaded(true)
+        setFailed(true)
+        onLoadError(err)
+    }
+
+    const loadSource = (index = 0) => {
+        if (!sources.length || !playerRef.current) return
+
+        setLoaded(false)
+        playerRef.current
+            .load(sources[index]?.src)
+            .then(() => {
+                setFailed(false)
+            })
+            .catch(err => {
+                console.error(err)
+                switch (err.code) {
+                    case ERROR_ANOTHER_LOAD:
+                        return
+                    case ERROR_NO_PLAYER:
+                        setTimeout(() => loadSource(index), 0)
+                        break
+                    default: {
+                        const nextIndex = sources.length > index + 1 && index + 1
+                        if (nextIndex) {
+                            setTimeout(() => loadSource(nextIndex), 0)
+                        } else {
+                            handleError(err)
+                        }
+                    }
+                }
+            })
+            .finally(() => {
+                setLoaded(true)
+            })
+    }
+
+    useEffect(() => {
+        const video = videoRef.current
+        const container = containerRef.current
+        if (!video || !container) return
+
+        video.volume = volume
+
+        shaka.polyfill.installAll() // todo: check if we need that
+
+        playerRef.current = new shaka.Player(video)
+        if (!playerRef.current) return
+
+        playerRef.current.addEventListener('error', handleError)
+        playerRef.current.addEventListener('buffering', (e) => {
+            setBuffering(e.buffering)
+        })
+
+        uiRef.current = new shaka.ui.Overlay(
+            playerRef.current,
+            container,
+            video,
+            reportText
+        )
+        uiRef.current.configure({
+            addBigPlayButton: true,
+            controlPanelElements: [],
+            addSeekBar: false
+        })
+
+        const { Report, TextSelection } = customControls(onReport)
+        shaka.ui.Controls.registerElement('report', new Report.Factory(reportText))
+        shaka.ui.Controls.registerElement('captions', new TextSelection.Factory())
+
+        loadSource()
+        hideSpinner(container)
+
+        // return () => {
```

Comment:
Shouldn't we destroy the elements on unmounted?

Reply:
we get errors when we do that, needs further investigation